core/src/main/java/org/elasticsearch/common/lucene/store/ByteArrayIndexInput.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.common.lucene.store;
+
+import org.apache.lucene.store.IndexInput;
+
+import java.io.EOFException;
+import java.io.IOException;
+
+/**
+ * Wraps array of bytes into IndexInput
+ */
+public class ByteArrayIndexInput extends IndexInput {
+    private final byte[] bytes;
+
+    private int pos;
+
+    private int offset;
+
+    private int length;
```

Comment:
As an aside, maybe we should consider changing our APIs to use `long` so we don't have to worry about byte arrays larger than ~2.14gb?